Gdk.EventKey is a pointer in Vala 0.23+
--- ibus-xkb-1.5.0.20140114/ui/gtk3/switcher.vala.orig	2014-01-11 15:22:55.000000000 +0100
+++ ibus-xkb-1.5.0.20140114/ui/gtk3/switcher.vala	2014-12-20 10:53:47.543298768 +0100
@@ -328,21 +328,21 @@
 
     public override bool key_press_event(Gdk.EventKey e) {
         bool retval = true;
-        Gdk.EventKey *pe = &e;
+        Gdk.EventKey pe = e;
 
         if (m_popup_delay_time > 0) {
             restore_window_position("pressed");
         }
 
         do {
-            uint modifiers = KeybindingManager.MODIFIER_FILTER & pe->state;
+            uint modifiers = KeybindingManager.MODIFIER_FILTER & pe.state;
 
             if ((modifiers != m_modifiers) &&
                 (modifiers != (m_modifiers | Gdk.ModifierType.SHIFT_MASK))) {
                 break;
             }
 
-            if (pe->keyval == m_keyval) {
+            if (pe.keyval == m_keyval) {
                 if (modifiers == m_modifiers)
                     next_engine();
                 else // modififers == m_modifiers | SHIFT_MASK
@@ -350,7 +350,7 @@
                 break;
             }
 
-            switch (pe->keyval) {
+            switch (pe.keyval) {
                 case 0x08fb: /* leftarrow */
                 case 0xff51: /* Left */
                     previous_engine();
@@ -366,7 +366,7 @@
                 case 0xff54: /* Down */
                     break;
                 default:
-                    debug("0x%04x", pe->keyval);
+                    debug("0x%04x", pe.keyval);
                     if (m_loop != null) {
                         m_loop.quit();
                         m_loop = null;
@@ -379,7 +379,7 @@
     }
 
     public override bool key_release_event(Gdk.EventKey e) {
-        Gdk.EventKey *pe = &e;
+        Gdk.EventKey pe = e;
 
         if (KeybindingManager.primary_modifier_still_pressed((Gdk.Event *)pe,
             m_primary_modifier)) {
